Turn/src/Store.cpp

```diff
@@ -3,13 +3,101 @@
 
 #include "..\include\Common.h"
 #include "..\include\Store.h"
+#include "..\include\ItemTypes.h"
 
 
 using namespace std;
 using namespace Common;
 
-void Store::StoreFront(Player* _Player){
-    ClearScreen();
-    cout << "Work in progress!" << endl;
-    system("PAUSE");
+const unsigned short arrowsCost = 200;
+const unsigned short bombCost = 100;
+const unsigned short potionCost = 150;
+const unsigned short whetstoneCost = 20;
+
+//Messages
+string lowCoinsMessage = " Insufficient Coins !";
+string invalidChoice = " Invalid Choice Entered !";
+
+void Store::StoreFront(Player* _Player) {
+
+	unsigned short choice = 0;
+	unsigned short coins;
+
+	do
+	{
+		ClearScreen();
+		coins = _Player->GetCoins();
+		cout << endl << " Jeremy's Tools Store\tPlayer: ";
+		cout << coins << " Coins" << endl << endl;
+		cout << " ----ITEM----------------COST---------------- " << endl << endl;
+		cout << " 1) Arrows x 5\t\t " << arrowsCost << endl;
+		cout << " 2) Bomb\t\t " << bombCost << endl;
+		cout << " 3) Potion\t\t " << potionCost << endl;
+		cout << " 4) Whetstone\t\t " << whetstoneCost << endl << endl;
+		cout << " 0) Exit" << endl << endl;
+		cout << " -------------------------------------------- " << endl << endl;
+		cout << " What do you want to buy today?" << endl << endl << " ";
+		cin >> choice;
+		cout << endl;
+		switch (choice)
+		{
+		case ITEMTYPE::ARROWS:
+			if (coins < arrowsCost)
+			{
+				cout << lowCoinsMessage << endl;
+			}
+			else
+			{
+				cout << " 5 Arrows purchased ! " << endl;
+				_Player->AddStoreItemToInventory(ITEMTYPE::ARROWS);
+				_Player->LoseCoins(arrowsCost);
+			}
+			Sleep(1000);
+			break;
+		case ITEMTYPE::BOMB:
+			if (coins < bombCost)
+			{
+				cout << lowCoinsMessage << endl;
+			}
+			else
+			{
+				cout << " Bomb purchased !" << endl;
+				_Player->AddStoreItemToInventory(ITEMTYPE::BOMB);
+				_Player->LoseCoins(bombCost);
+			}
+			Sleep(1000);
```

Comment:
Use SLEEP_MS instead of a magic number. Same thing for the other cases

Reply:
done all the changes suggested